homeassistant/components/roon/server.py

```diff
@@ -22,28 +22,33 @@ def __init__(self, hass, config_entry):
         self.config_entry = config_entry
         self.hass = hass
         self.roonapi = None
+        self.roon_id = None
         self.all_player_ids = set()
         self.all_playlists = []
         self.offline_devices = set()
         self._exit = False
         self._roon_name_by_id = {}
 
-    @property
-    def host(self):
-        """Return the host of this server."""
-        return self.config_entry.data[CONF_HOST]
-
     async def async_setup(self, tries=0):
-        """Set up a roon server based on host parameter."""
-        host = self.host
+        """Set up a roon server based on config parameters."""
         hass = self.hass
+        # Host will be None for configs using discovery
+        host = self.config_entry.data[CONF_HOST]
         token = self.config_entry.data[CONF_API_KEY]
-        _LOGGER.debug("async_setup: %s %s", token, host)
-        self.roonapi = RoonApi(ROON_APPINFO, token, host, blocking_init=False)
+        # Default to None for compatibility with older configs
+        core_id = self.config_entry.data.get(CONF_ROON_ID, None)
```

Comment:
`None` is the default.

```suggestion
        core_id = self.config_entry.data.get(CONF_ROON_ID)
```

Would it be possible to fetch the room ID and update the config entry during setup ?

Reply:
When discovery works properly - `host` isn't set - and the roon integration uses `roon_id` to keep track of which server it is connected to (and authorised by).

The discovery packet returned by roon devices includes this id - and the library can be efficient in reconnecting.

So we do need to store it.